src/main/java/com/splunk/logtosplunk/event_loggers/BlockEventLogger.java

```diff
@@ -0,0 +1,103 @@
+package com.splunk.logtosplunk.event_loggers;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import com.splunk.logtosplunk.LogToSplunkMod;
+import com.splunk.logtosplunk.SplunkMessagePreparer;
+import com.splunk.logtosplunk.actions.BlockEventAction;
+import com.splunk.logtosplunk.loggable_events.LoggableBlockEvent;
+
+import net.minecraft.block.Block;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.Vec3;
+import net.minecraft.world.World;
+import net.minecraftforge.event.world.BlockEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+/**
+ * Handles the logging of block events.
+ */
+public class BlockEventLogger {
+    private static final String LOG_NAME_MODIFIER = " - BLOCK";
+    private static final Logger logger = LogManager.getLogger(LogToSplunkMod.LOGGER_NAME + LOG_NAME_MODIFIER);
+
+    /**
+     * Prepares and forwards data to be sent to splunk.
+     */
+    private SplunkMessagePreparer messagePreparer;
+
+    /**
+     * Constructor.
+     *
+     * @param messagePreparer used to process this classes captured data.
+     */
+    public BlockEventLogger(SplunkMessagePreparer messagePreparer) {
+        this.messagePreparer = messagePreparer;
+    }
+
+    /**
+     * Captures Block BreakEvents and sends them to the message preparer.
+     *
+     * @param event The captured BreakEvent.
+     */
+    @SubscribeEvent
```

Comment:
On the other hand conditional statements that I have to write are harder to maintain and are more tightly coupled. I don't know how much of this factors in, but all of the subscriptions are handled server side, so if I had to guess, I think we'll be adding a minimum amount of overhead by using subscribers rather than a big conditional statement.  In the long term I'd like to be able to register even loggers via a conf file and I think that would be incompatible with hard coding our event loggers into a conditional statement.